src/util/samplebuffer.cpp

```diff
@@ -12,6 +11,17 @@ SampleBuffer::SampleBuffer(SINT size)
 
 SampleBuffer::~SampleBuffer() {
     SampleUtil::free(m_data);
+
+    /* cleanup the memory, just in case of use after free */
+    m_data = nullptr;
+    m_size = 0;
```

Comment:
Using it after destruct will crash anyway. Please remove that unnecessary code.

Reply:
No, it isn't true.
If you use it after destruct, it might just point to the heap and corrupt something else which will make it even harder to debug.

Reply:
Ok, than please add a comment to the code explaining the reason.

Reply:
I think there is one already, but I'll elaborate further more.

Reply:
I've added a longer explanation, if you disagree I won't discuss it further and simply remove the pointer reset.

Reply:
Still an unnecessary performance cost. This only makes sense from a practical standpoint, but accessing the buffer after the destructor is UB anyways so its still questionable whether this makes much sense from a theoretical perspective.